src/bootrom.rs

```diff
@@ -0,0 +1,79 @@
+use core::mem;
+
+// A bootrom function table code.
+pub type RomFnTableCode = [u8; 4];
+
+// Defines a code.
+macro_rules! table_codes {
+    ($($name:ident, $code:expr;) +) => {
+        $(pub const $name: RomFnTableCode = $code;) +
+    };
+}
+
+// TODO: wait for newer rust version where stringify works on bytes.
+table_codes! {
+    // 2.8.3.1.3. Flash Access Functions
+    IF, *b"\0\0IF";
+    EX, *b"\0\0EX";
+    RE, *b"\0\0RE";
+    RP, *b"\0\0RP";
+    FC, *b"\0\0FC";
+    CX, *b"\0\0CX";
+}
+
+// Create a ROMFuncs struct using the lookup function and the bootrom function's code.
+macro_rules! rom_funcs {
+    ($arg:expr, $lookup_func:expr, $($name:ident, $code:expr;) +) => {
+     ROMFuncs{
+         $($name: mem::transmute($lookup_func($arg, mem::transmute($code)))), +
+     }
+ }
+}
+
+macro_rules! ptr_def {
```

Comment:
You can omit the type in casts, so you can do this:

```rust
const ROM_TABLE_LOOKUP_PTR: *const u16 = 0x18 as _;
```

I think this is much more readable than introducing custom syntax with a macro

Reply:
Do you know why the `as _` is needed?

Reply:
Done and removed `ptr_def!` macro.